CRM/Utils/Hook.php

```diff
@@ -2785,4 +2785,21 @@ public static function alterCustomFieldDisplayValue(&$displayValue, $value, $ent
     );
   }
 
+  /**
+   * Alter APIv4 route permissions based on the Entity and Action
+   *
+   * @param array $permissions
+   * @param string $entity
+   * @param string $action
+   *
+   * @return mixed
+   */
+  public static function alterApiRoutePermissions(&$permissions, $entity, $action) {
```

Comment:
Could we put a comment on here to indicate that the hook is experimental?

My sense from prior discussion was (a) the existing/default guard on the API route is *likely* extraneous but (b) we're not confident enough to straight-up remove it. So the hook allows one to do a narrower carve-out. Makes sense.

However, at some point, we might make the bigger leap - and if we do that, then I don't think there should be a presumption that this will continue to fire.

Reply:
If it's just a step toward removing the permission check, maybe the hook approach is too complex and all we need is a whitelist?

Reply:
I am not really sure we want to be going the whole hog and removing the permission check, I just wanted to expose this one specific API publicly

Reply:
I can see it either way with straight removal or hook.It’s a trade between simplicity/maintainability and risk/security/unknowns. 

Hook+comments (“experimental...may be removed...”) at least mitigates maintainabilty concern on the hook side. Optionally, renaming from “hook_*” to “civi.*” could also signl